app/javascript/src/components/Dashboard/Contacts/Pane/Form.jsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from "react";
+
+import { Formik, Form } from "formik";
+import { Check } from "neetoicons";
+import { Button, Pane, Toastr } from "neetoui";
+import { Input, Select } from "neetoui/formik";
+
+import { CONTACT_FORM_VALIDATION_SCHEMA, ROLES_DATA } from "../constants";
+
+const ContactForm = ({ onClose, contact, isEdit }) => {
+  const [submitted, setSubmitted] = useState(false);
+
+  const handleSubmit = () => {
+    Toastr.success("Contact added successfully");
+    onClose();
+  };
+
+  return (
+    <Formik
+      initialValues={contact}
+      validateOnBlur={submitted}
+      validateOnChange={submitted}
+      validationSchema={CONTACT_FORM_VALIDATION_SCHEMA}
+      onSubmit={handleSubmit}
+    >
+      {({ isSubmitting }) => (
+        <Form className="w-full">
+          <Pane.Body className="space-y-6">
+            <div className="flex w-full justify-between space-x-4">
+              <Input
+                required
+                className="flex-grow-1"
+                label="First Name"
+                name="firstName"
+                placeholder="Enter first name"
+              />
+              <Input
+                required
+                className="flex-grow-1"
+                label="Last Name"
+                name="lastName"
+                placeholder="Enter last name"
+              />
+            </div>
+            <Input
+              required
+              className="w-full flex-grow-0"
+              label="Email"
+              name="email"
+              placeholder="Enter your email address"
+            />
+            <Select
+              isSearchable
+              required
+              className="w-full flex-grow-0"
+              label="Role"
+              name="role"
+              placeholder="Select Role"
+              options={ROLES_DATA.map(({ label, value }) => ({
+                label,
+                value,
+              }))}
+            />
+          </Pane.Body>
+          <Pane.Footer>
+            <Button
+              className="mr-3"
+              disabled={isSubmitting}
+              icon={Check}
+              label={isEdit ? "Update" : "Save Changes"}
+              loading={isSubmitting}
+              size="large"
+              style="primary"
+              type="submit"
+              onClick={() => setSubmitted(true)}
+            />
+            <Button
+              label="Cancel"
+              size="large"
+              style="text"
+              onClick={onClose}
+            />
```

Comment:
Go through the Slack thread.

Reply:
Hi. I had a doubt regarding this.

<img width="1282" alt="Screenshot 2022-09-28 at 1 02 53 PM" src="https://user-images.githubusercontent.com/44946937/192737728-8b3a3fac-41a4-4d18-a0c0-1fd1e654dea3.png">

The above Formik Cancel button type attribute doesn't seem to suit our use case as, our Cancel button is supposed to close the pane and not just reset the form values. So is it not okay to just handle it using an onClose function that would close the pane? Please do correct me if I'm wrong.

Reply:
@KrishnapriyaSkk If you have added the logic to `reset` the form values in the `onClose` function then it's fine to avoid the `reset` type. But in general, even if we provide the `reset` type along with the `onClick`, there's no harm.

Reply:
Okay got it. Thank you so much.